---
title: "Final_results"
author: "Akira Hirata"
date: "2024-05-04"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(dplyr)
library(readr)
library(venn)
library(gridExtra)
library(grid)
library(ggplot2)
library(lattice)
#install.packages("VennDiagram")
library(VennDiagram)
library(extrafont)
#install.packages("patchwork")
library(patchwork)
library(viridis)
library(RColorBrewer)
loadfonts()
```

# Number of elements per sample

## BQSR 
```{r, message=FALSE}
data <- read_csv("ME/ME_in_BQSR.csv")
data <- data[data$Individual != 24,]
tools <- unique(data$Tool)
plot_list <- list()
vcolors <- viridis(3)
type_colors <- c("ALU" = vcolors[1], "L1" = vcolors[2], "SVA" = vcolors[3])
for (tool in tools) {
  tool_data <- data[data$Tool == tool,]
  plot <- ggplot(data = tool_data, aes(x = Individual, fill = Type)) + geom_bar() + labs(title = paste("Tool:", tool), y = "Number of TEs") + theme_minimal() + scale_fill_manual(values = type_colors)
  plot_list[[tool]] <- plot
}
final_plot <- (plot_list[[1]] | plot_list[[2]] | plot_list[[3]]) + plot_annotation(title = "Figure S1: Number of TEs per Individual detected by each tool", caption = "Source: CVID patients")
print(final_plot)
ggsave(filename = "Final_plots/numberTEsBQSR.png", final_plot, width = 15, height = 7)
```
```{r, message=FALSE}
data <- read_csv("ME/ME_in_BQSR.csv")
tools <- unique(data$Tool)
plot_list <- list()
vcolors <- viridis(3)
type_colors <- c("ALU" = vcolors[1], "L1" = vcolors[2], "SVA" = vcolors[3])
for (tool in tools) {
  tool_data <- data[data$Tool == tool,]
  plot <- ggplot(data = tool_data, aes(x = Individual, fill = Type)) + geom_bar() + labs(title = paste("Tool:", tool), y = "Number of TEs") + theme_minimal() + scale_fill_manual(values = type_colors)
  plot_list[[tool]] <- plot
}
final_plot <- (plot_list[[1]] | plot_list[[2]] | plot_list[[3]]) + plot_annotation(title = "Figure S6: Number of TEs per Individual detected by each tool with S24", caption = "Source: CVID patients")
print(final_plot)
ggsave(filename = "Final_plots/numberTEsBQSR24.png", final_plot, width = 15, height = 7)
```

## Argentinian Trio
```{r, message=FALSE}
data <- read_csv("ME/ME_in_Family.csv")
unique_tools <- unique(data$Tool)
plot_list <- list()
vcolors <- viridis(3)
type_colors <- c("ALU" = vcolors[1], "L1" = vcolors[2], "SVA" = vcolors[3])
for (tool in unique_tools) {
  tool_data <- data[data$Tool == tool,]
  plot <- ggplot(data = tool_data, aes(x = Individual, fill = Type)) + geom_bar() + labs(title = paste("Tool:", tool), y = "Number of TEs") + theme_minimal() + scale_fill_manual(values = type_colors) + theme(axis.text.x = element_text(angle = 65, vjust = 1.25, hjust=0.85))
  plot_list[[tool]] <- plot
}
final_plot <- (plot_list[[1]] | plot_list[[2]] | plot_list[[3]]) + plot_annotation(title = "Figure S2: Number of TEs per Individual detected by each tool", caption = "Source: Argentinian Trio")
print(final_plot)
ggsave(filename = "Final_plots/numberTEsFamily.png", final_plot, width = 15, height = 7)
```

## Mediterranean Fever

```{r, message=FALSE}
data <- read_csv("ME/ME_in_fever.csv")
unique_tools <- unique(data$Tool)
plot_list <- list()
vcolors <- viridis(3)
type_colors <- c("ALU" = vcolors[1], "L1" = vcolors[2], "SVA" = vcolors[3])
for (tool in unique_tools) {
  tool_data <- data[data$Tool == tool,]
  plot <- ggplot(data = tool_data, aes(x = Individual, fill = Type)) + geom_bar() + labs(title = paste("Tool:", tool), y = "Number of TEs") + theme_minimal() + scale_fill_manual(values = type_colors) + theme(axis.text.x = element_text(angle = 65, vjust = 1.25, hjust=0.85))
  plot_list[[tool]] <- plot
}
final_plot <- (plot_list[[1]] | plot_list[[2]] | plot_list[[3]]) + plot_annotation(title = "Figure S3: Number of TEs per Individual detected by each tool", caption = "Source: Mediterranean Fever")
print(final_plot)
ggsave(filename = "Final_plots/numberTEsMEFV.png", final_plot, width = 15, height = 7)

```

## Vall d'Hebron WES

```{r, message=FALSE}
data <- read_csv("ME/ME_in_VHWES.csv")
unique_tools <- unique(data$Tool)
plot_list <- list()
vcolors <- viridis(3)
type_colors <- c("ALU" = vcolors[1], "L1" = vcolors[2], "SVA" = vcolors[3])
for (tool in unique_tools) {
  tool_data <- data[data$Tool == tool,]
  plot <- ggplot(data = tool_data, aes(x = Individual, fill = Type)) + geom_bar() + labs(title = paste("Tool:", tool), y = "Number of TEs") + theme_minimal() + scale_fill_manual(values = type_colors) + theme(axis.text.x = element_text(angle = 65, vjust = 1.25, hjust=0.85))
  plot_list[[tool]] <- plot
}
final_plot <- (plot_list[[1]] | plot_list[[2]] | plot_list[[3]]) + plot_annotation(title = "Figure S4: Number of TEs per Individual detected by each tool", caption = "Source: Vall d'Hebron WES")
print(final_plot)
ggsave(filename = "Final_plots/numberTEsVHWES.png", final_plot, width = 15, height = 7)
```

## Vall d'Hebron WGS

```{r, message=FALSE}
data <- read_csv("ME/ME_in_VH.csv")
unique_tools <- unique(data$Tool)
plot_list <- list()
vcolors <- viridis(3)
type_colors <- c("ALU" = vcolors[1], "L1" = vcolors[2], "SVA" = vcolors[3])
for (tool in unique_tools) {
  tool_data <- data[data$Tool == tool,]
  plot <- ggplot(data = tool_data, aes(x = Individual, fill = Type)) + geom_bar() + labs(title = paste("Tool:", tool), y = "Number of TEs") + theme_minimal() + scale_fill_manual(values = type_colors) + theme(axis.text.x = element_text(angle = 65, vjust = 1.25, hjust=0.85))
  plot_list[[tool]] <- plot
}
final_plot <- (plot_list[[1]] | plot_list[[2]] | plot_list[[3]]) + plot_annotation(title = "Figure S5: Number of TEs per Individual detected by each tool", caption = "Source: Vall d'Hebron WGS")
print(final_plot)
ggsave(filename = "Final_plots/numberTEsVHWGS.png", final_plot, width = 15, height = 7)
```

# IEI detection

## CVID patients
```{r, warning=FALSE}
gene_data <- read_csv("genes/genes_BQSR.csv") %>% distinct()
vcolors <- viridis(3)
type_colors <- c("melt" = vcolors[1], "mobster" = vcolors[2], "scramble" = vcolors[3])
gene_counts <- gene_data %>%
  group_by(Tool, Gene_name) %>%
  summarise(Count = n())

final_plot <- ggplot(gene_counts, aes(x = Tool, y = Count, fill = Tool, label = Gene_name)) +geom_bar(stat = "identity") +geom_text(position = position_stack(vjust = 0.5), size = 4) +labs(title = "Figure S7. Number of candidate genes (CVID) detected by each method", x = "Tool", y = "Count", caption = "Source: CVID patients") +theme_minimal() + scale_fill_manual(values = type_colors)
print(final_plot)
ggsave(filename = "Final_plots/CVID_candidate_genes.png", final_plot, width = 15, height = 7)
```

## Argentinian trio
```{r, warning=FALSE}
gene_data <- read_csv("genes/genes_Family.csv") %>% distinct()
gene_counts <- gene_data %>%
  group_by(Tool, Gene_name) %>%
  summarise(Count = n())

ggplot(gene_counts, aes(x = Tool, y = Count, fill = Tool, label = Gene_name)) +geom_bar(stat = "identity") +geom_text(color = "grey", position = position_stack(vjust = 0.5), size = 4) +labs(title = "Number of Genes detected by each method", x = "Tool", y = "Count") +theme_minimal() +scale_fill_viridis_d()
```

## Mediterranean fever
```{r, warning=FALSE}
gene_data <- read_csv("genes/genes_fever.csv") %>% distinct()
gene_counts <- gene_data %>%
  group_by(Tool, Gene_name) %>%
  summarise(Count = n())

ggplot(gene_counts, aes(x = Tool, y = Count, fill = Tool, label = Gene_name)) +geom_bar(stat = "identity") +geom_text(color = "grey", position = position_stack(vjust = 0.5), size = 4) +labs(title = "Number of Genes detected by each method", x = "Tool", y = "Count") +theme_minimal() +scale_fill_viridis_d()
```

## Vall d'Hebron WES

```{r, warning=FALSE}
gene_data <- read_csv("genes/genes_VHWES.csv") %>% distinct()
vcolors <- viridis(3)
type_colors <- c("melt" = vcolors[1], "mobster" = vcolors[2], "scramble" = vcolors[3])
gene_counts <- gene_data %>%
  group_by(Tool, Gene_name) %>%
  summarise(Count = n())

final_plot <- ggplot(gene_counts, aes(x = Tool, y = Count, fill = Tool, label = Gene_name)) +geom_bar(stat = "identity") +geom_text(position = position_stack(vjust = 0.5), size = 4) +labs(title = "Figure S8. Number of candidate genes (IEI) detected by each method", x = "Tool", y = "Count", caption = "Source: Vall d'Hebron WES") +theme_minimal() + scale_fill_manual(values = type_colors)
print(final_plot)
ggsave(filename = "Final_plots/IEI_WES_candidate_genes.png", final_plot, width = 15, height = 7)
```

## Vall d'Hebron WGS

```{r, warning=FALSE}
gene_data <- read_csv("genes/genes_VH.csv") %>% distinct()
vcolors <- viridis(3)
type_colors <- c("melt" = vcolors[1], "mobster" = vcolors[2], "scramble" = vcolors[3])
gene_counts <- gene_data %>%
  group_by(Tool, Gene_name) %>%
  summarise(Count = n())

final_plot <- ggplot(gene_counts, aes(x = Tool, y = Count, fill = Tool, label = Gene_name)) +geom_bar(stat = "identity")  +labs(title = "Figure S9. Number of candidate genes (IEI) detected by each method", x = "Tool", y = "Count", caption = "Source: Vall d'Hebron WGS") +theme_minimal() + scale_fill_manual(values = type_colors)
print(final_plot)
ggsave(filename = "Final_plots/IEI_WGS_candidate_genes.png", final_plot, width = 15, height = 7)

melt <- gene_data[gene_data$Tool=="melt",]
mobster <- gene_data[gene_data$Tool=="mobster",]
scramble <- gene_data[gene_data$Tool=="scramble",]
set1 <- paste(melt$Individual,melt$Chromosome,melt$Gene_name, sep="_")
set2 <- paste(mobster$Individual,mobster$Chromosome,mobster$Gene_name, sep="_")
set3 <- paste(scramble$Individual,scramble$Chromosome,scramble$Gene_name, sep="_")
common_results_initial <- Reduce(intersect, list(set1, set2, set3i))
library(RColorBrewer)
myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'IEI_VH_comparison.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)

melt_inside <- melt[melt$Distance_tag=="Close",]
mobster_inside <- mobster[mobster$Distance_tag=="Close",]
scramble_inside <- scramble[scramble$Distance_tag=="Close",]
set1i <- paste(melt_inside$Individual,melt_inside$Chromosome,melt_inside$Gene_name, sep="_")
set2i <- paste(mobster_inside$Individual,mobster_inside$Chromosome,mobster_inside$Gene_name, sep="_")
set3i <- paste(scramble_inside$Individual,scramble_inside$Chromosome,scramble_inside$Gene_name, sep="_")
common_results <- Reduce(intersect, list(set1i, set2i, set3i))
```


# Method comparison cohort (MELT vs SCRAMble vs mobster)

## CVID patients
```{r}
melt <- read_csv("melt/exp_obs_gene_BQSR_melt.csv") %>% distinct()
melt <- melt[!melt$Sample %in% c(24),]
melt <- melt[,1:7]
mobster <- read_csv("mobster/exp_obs_gene_BQSR_mobster.csv") %>% distinct() 
mobster <- mobster[!mobster$Sample %in% c(24),]
mobster <- mobster[,1:7]
scramble <- read_csv("scramble/exp_obs_gene_BQSR_scramble.csv") %>% distinct() 
scramble <- scramble[!scramble$Sample %in% c(24),]
scramble <- scramble[,1:7]
set1 <- paste(melt$Sample,melt$Chromosome,melt$Gene, sep="_")
set2 <- paste(mobster$Sample,mobster$Chromosome,mobster$Gene, sep="_")
set3 <- paste(scramble$Sample,scramble$Chromosome,scramble$Gene, sep="_")

library(RColorBrewer)
myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'CVID_comparison.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)
common_results <- Reduce(intersect, list(melt, mobster, scramble))
melt$Method <- "Melt"
mobster$Method <- "Mobster"
scramble$Method <- "Scramble"
combined_data <- bind_rows(melt, mobster, scramble)
vcolors <- viridis(3)
type_colors <- c("Melt" = vcolors[1], "Mobster" = vcolors[2], "Scramble" = vcolors[3])
gene_counts <- combined_data %>%group_by(Sample, Method) %>% summarise(Count = n_distinct(Gene))
final_plot <- ggplot(data = gene_counts, aes(x = Sample, y = Count, fill = Method)) + geom_bar(stat = "identity", position = "dodge") +theme_minimal() + scale_fill_manual(values = type_colors) +labs(title="Figure S10. Number of TEs in CVID patients per tool",y = "Number of TEs",caption = "Source: CVID patients",fill = "Method")
print(final_plot)
ggsave(filename = "Final_plots/CVIDcomparison.png", final_plot, width = 15, height = 7)
```

```{r}
tools <- unique(combined_data$Method)
plot_list <- list()
vcolors <- viridis(5)
combined_data$Distance_Tag <- factor(combined_data$Distance_Tag, levels = c('Inside', 'Close', 'Near', 'Far', 'Very Far'))
type_colors <- c("Inside" = vcolors[1], "Close" = vcolors[2], "Near" = vcolors[3],  "Far" = vcolors[4],  "Very Far" = vcolors[5])
for (tool in tools) {
  tool_data <- combined_data[combined_data$Method == tool,]
  plot <- ggplot(data = tool_data, aes(x = Sample, fill = Distance_Tag)) + geom_bar() + labs(title = paste("Tool:", tool), y = "Number of TEs") + theme_minimal() + scale_fill_manual(values = type_colors)
  plot_list[[tool]] <- plot
}
final_plot <- (plot_list[[1]] | plot_list[[2]] | plot_list[[3]]) + plot_annotation(title = "Figure S1: Number of TEs per Individual detected by each tool", caption = "Source: CVID patients")
print(final_plot)
```

## Argentinian trio

```{r}
melt <- read_csv("melt/exp_obs_gene_Family_melt.csv") %>% distinct()
melt <- melt[,1:7]
mobster <- read_csv("mobster/exp_obs_gene_Family_mobster.csv") %>% distinct() 
mobster <- mobster[,1:7]
scramble <- read_csv("scramble/exp_obs_gene_Family_scramble.csv") %>% distinct() 
scramble <- scramble[,1:7]
set1 <- paste(melt$Sample,melt$Chromosome,melt$Gene, sep="_")
set2 <- paste(mobster$Sample,mobster$Chromosome,mobster$Gene, sep="_")
set3 <- paste(scramble$Sample,scramble$Chromosome,scramble$Gene, sep="_")

myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'Argentinian_trio_comparison.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)
common_results <- Reduce(intersect, list(melt, mobster, scramble))
melt$Method <- "Melt"
mobster$Method <- "Mobster"
scramble$Method <- "Scramble"
combined_data <- bind_rows(melt, mobster, scramble)
vcolors <- viridis(3)
type_colors <- c("Melt" = vcolors[1], "Mobster" = vcolors[2], "Scramble" = vcolors[3])
gene_counts <- combined_data %>%group_by(Sample, Method) %>% summarise(Count = n_distinct(Gene))
final_plot <- ggplot(data = gene_counts, aes(x = Sample, y = Count, fill = Method)) + geom_bar(stat = "identity", position = "dodge") +theme_minimal() + scale_fill_manual(values = type_colors) +labs(title="Figure S11. Number of TEs in Argentinian trio per tool",y = "Number of TEs",caption = "Source: Argentinian trio",fill = "Method")
print(final_plot)
ggsave(filename = "Final_plots/Familycomparison.png", final_plot, width = 15, height = 7)
```

## Mediterranean fever

```{r}
melt <- read_csv("melt/exp_obs_gene_fever_melt.csv") %>% distinct()
mobster <- read_csv("mobster/exp_obs_gene_fever_mobster.csv") %>% distinct() 
scramble <- read_csv("scramble/exp_obs_gene_fever_scramble.csv") %>% distinct() 
melt <- melt[,1:7]
mobster <- mobster[,1:7]
scramble <- scramble[,1:7]
set1 <- paste(melt$Sample,melt$Chromosome,melt$Gene, sep="_")
set2 <- paste(mobster$Sample,mobster$Chromosome,mobster$Gene, sep="_")
set3 <- paste(scramble$Sample,scramble$Chromosome,scramble$Gene, sep="_")

library(RColorBrewer)
myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'fever_comparison.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)
common_results <- Reduce(intersect, list(set1, set2, set3))
melt$Method <- "Melt"
mobster$Method <- "Mobster"
scramble$Method <- "Scramble"
combined_data <- bind_rows(melt, mobster, scramble)
vcolors <- viridis(3)
type_colors <- c("Melt" = vcolors[1], "Mobster" = vcolors[2], "Scramble" = vcolors[3])
gene_counts <- combined_data %>%group_by(Sample, Method) %>% summarise(Count = n_distinct(Gene))
final_plot <- ggplot(data = gene_counts, aes(x = Sample, y = Count, fill = Method)) + geom_bar(stat = "identity", position = "dodge") +theme_minimal() + scale_fill_manual(values = type_colors) +labs(title="Figure S12. Number of TEs in Mediterranean Fever patients per tool",y = "Number of TEs",caption = "Source: Mediterranean fever",fill = "Method") + theme(axis.text.x = element_text(angle = 65, vjust = 1.25, hjust=0.85))
print(final_plot)
ggsave(filename = "Final_plots/fevercomparison.png", final_plot, width = 15, height = 7)
```

## Vall d'Hebron WES

```{r}
melt <- read_csv("melt/exp_obs_gene_VHWES_melt.csv") %>% distinct()
mobster <- read_csv("mobster/exp_obs_gene_VHWES_mobster.csv") %>% distinct() 
scramble <- read_csv("scramble/exp_obs_gene_VH_WES_scramble.csv") %>% distinct() 
melt <- melt[,1:7]
mobster <- mobster[,1:7]
scramble <- scramble[,1:7]
set1 <- paste(melt$Sample,melt$Chromosome,melt$Gene, sep="_")
set2 <- paste(mobster$Sample,mobster$Chromosome,mobster$Gene, sep="_")
set3 <- paste(scramble$Sample,scramble$Chromosome,scramble$Gene, sep="_")

library(RColorBrewer)
myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'VHWES_comparison.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)
common_results <- Reduce(intersect, list(melt, mobster, scramble))
melt$Method <- "Melt"
mobster$Method <- "Mobster"
scramble$Method <- "Scramble"
combined_data <- bind_rows(melt, mobster, scramble)
vcolors <- viridis(3)
type_colors <- c("Melt" = vcolors[1], "Mobster" = vcolors[2], "Scramble" = vcolors[3])
gene_counts <- combined_data %>%group_by(Sample, Method) %>% summarise(Count = n_distinct(Gene))
final_plot <- ggplot(data = gene_counts, aes(x = Sample, y = Count, fill = Method)) + geom_bar(stat = "identity", position = "dodge") +theme_minimal() + scale_fill_manual(values = type_colors) +labs(title="Figure S13. Number of TEs in Vall d'Hebron WES data per tool",y = "Number of TEs",caption = "Source: Vall d'Hebron WES",fill = "Method") + theme(axis.text.x = element_text(angle = 65, vjust = 1.25, hjust=0.85))
print(final_plot)
ggsave(filename = "Final_plots/VHWEScomparison.png", final_plot, width = 15, height = 7)
```

## Vall d'Hebron WGS

```{r}
melt <- read_csv("melt/exp_obs_gene_VH_melt.csv") %>% distinct()
mobster <- read_csv("mobster/exp_obs_gene_VH_mobster.csv") %>% distinct() 
scramble <- read_csv("scramble/exp_obs_gene_VH_scramble.csv") %>% distinct() 
melt <- melt[,1:7]
mobster <- mobster[,1:7]
scramble <- scramble[,1:7]
set1 <- paste(melt$Sample,melt$Chromosome,melt$Gene, sep="_")
set2 <- paste(mobster$Sample,mobster$Chromosome,mobster$Gene, sep="_")
set3 <- paste(scramble$Sample,scramble$Chromosome,scramble$Gene, sep="_")

library(RColorBrewer)
myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'VH_comparison.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)
common_results <- Reduce(intersect, list(melt, mobster, scramble))
melt$Method <- "Melt"
mobster$Method <- "Mobster"
scramble$Method <- "Scramble"
combined_data <- bind_rows(melt, mobster, scramble)
vcolors <- viridis(3)
type_colors <- c("Melt" = vcolors[1], "Mobster" = vcolors[2], "Scramble" = vcolors[3])
gene_counts <- combined_data %>%group_by(Sample, Method) %>% summarise(Count = n_distinct(Gene))
final_plot <- ggplot(data = gene_counts, aes(x = Sample, y = Count, fill = Method)) + geom_bar(stat = "identity", position = "dodge") +theme_minimal() + scale_fill_manual(values = type_colors) +labs(title="Figure S14. Number of TEs in Vall d'Hebron WGS data per tool",y = "Number of TEs",caption = "Source: Vall d'Hebron WGS",fill = "Method") + theme(axis.text.x = element_text(angle = 65, vjust = 1.25, hjust=0.85))
print(final_plot)
ggsave(filename = "Final_plots/VHWGScomparison.png", final_plot, width = 15, height = 7)
```

# Method comparison individual (MELT vs SCRAMble vs mobster)

## CVID patients

```{r}
melt <- read_csv("melt/exp_obs_gene_BQSR_melt.csv") %>% distinct()
melt <- melt[!melt$Sample %in% c(24),]
mobster <- read_csv("mobster/exp_obs_gene_BQSR_mobster.csv") %>% distinct() 
mobster <- mobster[!mobster$Sample %in% c(24),]
scramble <- read_csv("scramble/exp_obs_gene_BQSR_scramble.csv") %>% distinct() 
scramble <- scramble[!scramble$Sample %in% c(24),]
unique_datasets <- unique(melt$Sample)
for (dataset_id in unique_datasets) {
  set1 <- paste(melt$Chromosome[melt$Sample == dataset_id], melt$Gene[melt$Sample == dataset_id], sep="_")
  set2 <- paste(mobster$Chromosome[mobster$Sample == dataset_id], mobster$Gene[mobster$Sample == dataset_id], sep="_")
  set3 <- paste(scramble$Chromosome[scramble$Sample == dataset_id], scramble$Gene[scramble$Sample == dataset_id], sep="_")
  library(RColorBrewer)
  myCol <- brewer.pal(3, "Pastel2")
  venn.diagram(
    x = list(set1, set2, set3),
    category.names = c("MELT" , "mobster" , "SCRAMble"),
    filename = paste0('CVID_comparison_', dataset_id, '.png'),
    output = TRUE,
    imagetype = "png",
    height = 480, 
    width = 480, 
    resolution = 300,
    compression = "lzw",
    lwd = 2,
    lty = 'blank',
    fill = myCol,
    cex = .6,
    fontface = "bold",
    fontfamily = "sans",
    cat.cex = 0.6,
    cat.fontface = "bold",
    cat.default.pos = "outer",
    cat.pos = c(-27, 27, 135),
    cat.dist = c(0.055, 0.055, 0.085),
    cat.fontfamily = "sans",
    rotation = 1
  )
}
```

## Argentinian trio

```{r}
melt <- read_csv("melt/exp_obs_gene_Family_melt.csv") %>% distinct()
mobster <- read_csv("mobster/exp_obs_gene_Family_mobster.csv") %>% distinct() 
scramble <- read_csv("scramble/exp_obs_gene_Family_scramble.csv") %>% distinct() 
unique_datasets <- unique(melt$Sample)
for (dataset_id in unique_datasets) {
  set1 <- paste(melt$Chromosome[melt$Sample == dataset_id], melt$Gene[melt$Sample == dataset_id], sep="_")
  set2 <- paste(mobster$Chromosome[mobster$Sample == dataset_id], mobster$Gene[mobster$Sample == dataset_id], sep="_")
  set3 <- paste(scramble$Chromosome[scramble$Sample == dataset_id], scramble$Gene[scramble$Sample == dataset_id], sep="_")
  library(RColorBrewer)
  myCol <- brewer.pal(3, "Pastel2")
  venn.diagram(
    x = list(set1, set2, set3),
    category.names = c("MELT" , "mobster" , "SCRAMble"),
    filename = paste0('Comparison_', dataset_id, '.png'),
    output = TRUE,
    
    # Output features
    imagetype = "png",
    height = 480, 
    width = 480, 
    resolution = 300,
    compression = "lzw",
    
    # Circles
    lwd = 2,
    lty = 'blank',
    fill = myCol,
    
    # Numbers
    cex = .6,
    fontface = "bold",
    fontfamily = "sans",
    
    # Set names
    cat.cex = 0.6,
    cat.fontface = "bold",
    cat.default.pos = "outer",
    cat.pos = c(-27, 27, 135),
    cat.dist = c(0.055, 0.055, 0.085),
    cat.fontfamily = "sans",
    rotation = 1
  )
}
```

## Mediterranean fever

```{r}
melt <- read_csv("melt/exp_obs_gene_fever_melt.csv") %>% distinct()
mobster <- read_csv("mobster/exp_obs_gene_fever_mobster.csv") %>% distinct() 
scramble <- read_csv("scramble/exp_obs_gene_fever_scramble.csv") %>% distinct() 
unique_datasets <- unique(melt$Sample)
for (dataset_id in unique_datasets) {
  set1 <- paste(melt$Chromosome[melt$Sample == dataset_id], melt$Gene[melt$Sample == dataset_id], sep="_")
  set2 <- paste(mobster$Chromosome[mobster$Sample == dataset_id], mobster$Gene[mobster$Sample == dataset_id], sep="_")
  set3 <- paste(scramble$Chromosome[scramble$Sample == dataset_id], scramble$Gene[scramble$Sample == dataset_id], sep="_")
  library(RColorBrewer)
  myCol <- brewer.pal(3, "Pastel2")
  venn.diagram(
    x = list(set1, set2, set3),
    category.names = c("MELT" , "mobster" , "SCRAMble"),
    filename = paste0('Comparison_', dataset_id, '.png'),
    output = TRUE,
    
    # Output features
    imagetype = "png",
    height = 480, 
    width = 480, 
    resolution = 300,
    compression = "lzw",
    
    # Circles
    lwd = 2,
    lty = 'blank',
    fill = myCol,
    
    # Numbers
    cex = .6,
    fontface = "bold",
    fontfamily = "sans",
    
    # Set names
    cat.cex = 0.6,
    cat.fontface = "bold",
    cat.default.pos = "outer",
    cat.pos = c(-27, 27, 135),
    cat.dist = c(0.055, 0.055, 0.085),
    cat.fontfamily = "sans",
    rotation = 1
  )
}
```

## Vall d'Hebron WES

```{r}
melt <- read_csv("melt/exp_obs_gene_VHWES_melt.csv") %>% distinct()
mobster <- read_csv("mobster/exp_obs_gene_VHWES_mobster.csv") %>% distinct() 
scramble <- read_csv("scramble/exp_obs_gene_VH_WES_scramble.csv") %>% distinct() 
unique_datasets <- unique(mobster$Sample)
for (dataset_id in unique_datasets) {
  set1 <- paste(melt$Chromosome[melt$Sample == dataset_id], melt$Gene[melt$Sample == dataset_id], sep="_")
  set2 <- paste(mobster$Chromosome[mobster$Sample == dataset_id], mobster$Gene[mobster$Sample == dataset_id], sep="_")
  set3 <- paste(scramble$Chromosome[scramble$Sample == dataset_id], scramble$Gene[scramble$Sample == dataset_id], sep="_")
  library(RColorBrewer)
  myCol <- brewer.pal(3, "Pastel2")
  venn.diagram(
    x = list(set1, set2, set3),
    category.names = c("MELT" , "mobster" , "SCRAMble"),
    filename = paste0('Comparison_', dataset_id, '.png'),
    output = TRUE,
    
    # Output features
    imagetype = "png",
    height = 480, 
    width = 480, 
    resolution = 300,
    compression = "lzw",
    
    # Circles
    lwd = 2,
    lty = 'blank',
    fill = myCol,
    
    # Numbers
    cex = .6,
    fontface = "bold",
    fontfamily = "sans",
    
    # Set names
    cat.cex = 0.6,
    cat.fontface = "bold",
    cat.default.pos = "outer",
    cat.pos = c(-27, 27, 135),
    cat.dist = c(0.055, 0.055, 0.085),
    cat.fontfamily = "sans",
    rotation = 1
  )
}
```

## Vall d'Hebron WGS

```{r}
melt <- read_csv("melt/exp_obs_gene_VH_melt.csv") %>% distinct()
mobster <- read_csv("mobster/exp_obs_gene_VH_mobster.csv") %>% distinct() 
scramble <- read_csv("scramble/exp_obs_gene_VH_scramble.csv") %>% distinct() 
unique_datasets <- unique(melt$Sample)
for (dataset_id in unique_datasets) {
  set1 <- paste(melt$Chromosome[melt$Sample == dataset_id], melt$Gene[melt$Sample == dataset_id], sep="_")
  set2 <- paste(mobster$Chromosome[mobster$Sample == dataset_id], mobster$Gene[mobster$Sample == dataset_id], sep="_")
  set3 <- paste(scramble$Chromosome[scramble$Sample == dataset_id], scramble$Gene[scramble$Sample == dataset_id], sep="_")
  library(RColorBrewer)
  myCol <- brewer.pal(3, "Pastel2")
  venn.diagram(
    x = list(set1, set2, set3),
    category.names = c("MELT" , "mobster" , "SCRAMble"),
    filename = paste0('Comparison_', dataset_id, '.png'),
    output = TRUE,
    
    # Output features
    imagetype = "png",
    height = 480, 
    width = 480, 
    resolution = 300,
    compression = "lzw",
    
    # Circles
    lwd = 2,
    lty = 'blank',
    fill = myCol,
    
    # Numbers
    cex = .6,
    fontface = "bold",
    fontfamily = "sans",
    
    # Set names
    cat.cex = 0.6,
    cat.fontface = "bold",
    cat.default.pos = "outer",
    cat.pos = c(-27, 27, 135),
    cat.dist = c(0.055, 0.055, 0.085),
    cat.fontfamily = "sans",
    rotation = 1
  )
}
```

# Data comparion (WES vs WGS vs RNAseq)

## Individual S58/VH17i

```{r,message=FALSE}
analyze_data <- function(df1_path, df2_path, sample1, sample2, method_name) {
  df1 <- read_csv(df1_path) %>% distinct() 
  df1_extra <- df1[df1$Sample == sample1,]
  df2 <- read_csv(df2_path) %>% distinct()
  df2_extra <- df2[df2$Sample == sample2,]
  common_rows <- semi_join(df1_extra, df2_extra, by = c("Chromosome","Gene"))
  
  venn_diagram <- draw.pairwise.venn(
    area1 = nrow(df1_extra),                      
    area2 = nrow(df2_extra),                     
    cross.area = nrow(common_rows),                      
    category = c("WES data", "WGS data"),                 
    fill = c("lightblue", "pink"),       
    lty = "blank",                              
    cat.pos = c(-20, 20),                     
    scaled = TRUE,                          
    cat.fontface = "bold",
    output = TRUE 
  )
  
  not_common_rows_WES <- anti_join(df1_extra, df2_extra, by = c("Chromosome","Gene"))
  not_common_rows_WES$Distance_Tag <- factor(not_common_rows_WES$Distance_Tag, levels = c('Inside', 'Close', 'Near', 'Far', 'Very Far'))
  plot1 <- ggplot(data = not_common_rows_WES, mapping = aes(x = Sample, fill = Distance_Tag)) + geom_bar() + theme_minimal() + scale_fill_viridis_d() + labs(title = paste0("Type of distance individual ", sample1, "/", sample2, " ", method_name), y = "Nºdistance tag per sample", caption = "Source: Vall d'Hebron data WES", fill="Distance Tag")
  
  not_common_rows <- anti_join(df2_extra, df1_extra, by = c("Chromosome","Gene"))
  not_common_rows$Distance_Tag <- factor(not_common_rows$Distance_Tag, levels = c('Inside', 'Close', 'Near', 'Far', 'Very Far'))
  plot2 <- ggplot(data = not_common_rows, mapping = aes(x = Sample, fill = Distance_Tag)) + geom_bar() + theme_minimal() + scale_fill_viridis_d() + labs(y = "Nºdistance tag per sample", caption = "Source: Vall d'Hebron data WGS", fill="Distance Tag")
  
  filename <- paste0("Final_plots/comparison_", method_name, "_", sample1, "_", sample2, ".png")
  venn_filename <- paste0("Final_plots/Venn_", method_name, "_", sample1, "_", sample2, ".png")
  
  return(list(plots = list(plot1, plot2), venn_diagram = venn_diagram, filename = filename, venn_filename = venn_filename))
}

save_plots <- function(plot_list, filename) {
  final_plot <- arrangeGrob(grobs = plot_list, nrow = 1)
  ggsave(filename = filename, plot = final_plot, width = 15, height = 7)
}

save_venn <- function(venn_diagram, filename) {
  png(filename)
  grid.draw(venn_diagram)
  dev.off()
}

result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "S58", "VH17i", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv", "S58", "VH17i", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "S58", "VH17i", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)
```

## Individual S59/VH17f

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "S59" , "VH17f", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv",  "S59" , "VH17f", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv",  "S59" , "VH17f", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)
```

## Individual S60/VH17m

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "S60" , "VH17m", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv",  "S60" , "VH17m", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv",  "S60" , "VH17m", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)
```

## Individual VH60/VH11s

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "VH60" , "VH11s", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv", "VH60" , "VH11s", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "VH60" , "VH11s", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)
```


## Individual VH61/VH11i

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv","VH61" , "VH11i", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv", "VH61" , "VH11i", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "VH61" , "VH11i", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)


```

## Individual VH63/VH11f

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv","VH63" , "VH11f", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv", "VH63" , "VH11f", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "VH63" , "VH11f", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)

```

## Individual VH71/VH16i

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv","VH71" , "VH16i", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv", "VH71" , "VH16i", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "VH71" , "VH16i", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)
```


## Individual VH72/VH16m

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "VH72" , "VH16m", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv",  "VH72" , "VH16m", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv",  "VH72" , "VH16m", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)

```

## Individual VH73/VH16f

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "VH73" , "VH16f", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv",  "VH73" , "VH16f", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv",  "VH73" , "VH16f", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)

```

## Individual VH77/VH01i

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "VH77" , "VH01i", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv",  "VH77" , "VH01i", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "VH77" , "VH01i", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)
```

## Individual VH78/VH01m

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "VH78" , "VH01m", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv",  "VH78" , "VH01m", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "VH78" , "VH01m", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)

```

## Individual VH79/VH01f

```{r,message=FALSE}
result_melt <- analyze_data("melt/exp_obs_gene_VHWES_melt.csv", "melt/exp_obs_gene_VH_melt.csv", "VH79" , "VH01f", "melt")
save_plots(result_melt$plots, result_melt$filename)
save_venn(result_melt$venn_diagram, result_melt$venn_filename)
result_mobster <- analyze_data("mobster/exp_obs_gene_VHWES_mobster.csv", "mobster/exp_obs_gene_VH_mobster.csv",  "VH79" , "VH01f", "mobster")
save_plots(result_mobster$plots, result_mobster$filename)
save_venn(result_mobster$venn_diagram, result_mobster$venn_filename)
result_scramble <- analyze_data("scramble/exp_obs_gene_VH_WES_scramble.csv", "scramble/exp_obs_gene_VH_scramble.csv", "VH79" , "VH01f", "SCRAMble")
save_plots(result_scramble$plots, result_scramble$filename)
save_venn(result_scramble$venn_diagram, result_scramble$venn_filename)

```

# De novo insertions

## Vall d'Hebron Genome 
```{r}
melt <- read_csv("de_novo/de_novo_VH_melt.csv") %>% distinct()
mobster <- read_csv("de_novo/de_novo_VH_mobster.csv") %>% distinct() 
scramble <- read_csv("de_novo/de_novo_VH_scramble.csv") %>% distinct() 
set1 <- paste(melt$Family,melt$Gene, sep="_")
set2 <- paste(mobster$Family,mobster$Gene, sep="_")
set3 <- paste(scramble$Family,scramble$Gene, sep="_")

library(RColorBrewer)
myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'de_novo_comparison.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)
```
```{r,message=FALSE}
melt <- read_csv("de_novo/de_novo_VH_melt.csv") %>% distinct() %>% na.omit()
mobster <- read_csv("de_novo/de_novo_VH_mobster.csv") %>% distinct() %>% na.omit()
scramble <- read_csv("de_novo/de_novo_VH_scramble.csv") %>% distinct() %>% na.omit()
common_results <- Reduce(intersect, list(melt, mobster, scramble))

melt$Method <- "Melt"
mobster$Method <- "Mobster"
scramble$Method <- "Scramble"
combined_data <- bind_rows(melt, mobster, scramble)
vcolors <- viridis(3)
type_colors <- c("Melt" = vcolors[1], "Mobster" = vcolors[2], "Scramble" = vcolors[3])
gene_counts <- combined_data %>%group_by(Family, Method) %>% summarise(Count = n_distinct(Gene))
denovo_plot <- ggplot(data = gene_counts, aes(x = Family, y = Count, fill = Method)) + geom_bar(stat = "identity", position = "dodge") +theme_minimal() + scale_fill_manual(values = type_colors) +labs(title="Figure S28. Number of De novo insertions detected in WGS trios per tool",y = "Number of De novo detected",caption = "Source: Vall d'Hebron WGS",fill = "Method")
print(denovo_plot)
ggsave(filename = "Final_plots/denovoWGS.png",denovo_plot, width = 15, height = 7)
denovogenes_plot <- ggplot(data = common_results, mapping = aes(x = Family, fill = Gene)) + geom_bar() + theme_minimal() + scale_fill_viridis_d() + labs(title="Figure S29. Genes from De novo insertion detected in WGS trios",y = "Number of De novo detected", caption = "Source: Vall d'Hebron WGS", fill = "Genes")
print(denovogenes_plot)
ggsave(filename = "Final_plots/denovogenesWGS.png",denovogenes_plot, width = 15, height = 7)
```
## Vall d'Hebron Exome
```{r}
melt <- read_csv("de_novo/de_novo_VHWES_melt.csv") %>% distinct()
mobster <- read_csv("de_novo/de_novo_VHWES_mobster.csv") %>% distinct() 
scramble <- read_csv("de_novo/de_novo_VHWES_scramble.csv") %>% distinct() 
set1 <- paste(melt$Family,melt$Gene, sep="_")
set2 <- paste(mobster$Family,mobster$Gene, sep="_")
set3 <- paste(scramble$Family,scramble$Gene, sep="_")

library(RColorBrewer)
myCol <- brewer.pal(3, "Pastel2")
venn.diagram(
        x = list(set1, set2, set3),
        category.names = c("MELT" , "mobster" , "SCRAMble"),
        filename = 'de_novo_comparison_WES.png',
        output=TRUE,
        
        # Output features
        imagetype="png" ,
        height = 480 , 
        width = 480 , 
        resolution = 300,
        compression = "lzw",
        
        # Circles
        lwd = 2,
        lty = 'blank',
        fill = myCol,
        
        # Numbers
        cex = .6,
        fontface = "bold",
        fontfamily = "sans",
        
        # Set names
        cat.cex = 0.6,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.pos = c(-27, 27, 135),
        cat.dist = c(0.055, 0.055, 0.085),
        cat.fontfamily = "sans",
        rotation = 1
)
```

```{r,message=FALSE}
melt <- read_csv("de_novo/de_novo_VHWES_melt.csv") %>% distinct()
mobster <- read_csv("de_novo/de_novo_VHWES_mobster.csv") %>% distinct() 
scramble <- read_csv("de_novo/de_novo_VHWES_scramble.csv") %>% distinct() 
common_results <- Reduce(intersect, list(melt, mobster, scramble))

melt$Method <- "Melt"
mobster$Method <- "Mobster"
scramble$Method <- "Scramble"
combined_data <- bind_rows(melt, mobster, scramble)
vcolors <- viridis(3)
type_colors <- c("Melt" = vcolors[1], "Mobster" = vcolors[2], "Scramble" = vcolors[3])
gene_counts <- combined_data %>%group_by(Family, Method) %>% summarise(Count = n_distinct(Gene))
denovoWES_plot <- ggplot(data = gene_counts, aes(x = Family, y = Count, fill = Method)) + geom_bar(stat = "identity", position = "dodge") +theme_minimal() + scale_fill_manual(values = type_colors) +labs(title="Figure S30. Number of De novo insertions detected in WES trios per tool",y = "Number of De novo detected",caption = "Source: Vall d'Hebron WES",fill = "Method")
print(denovoWES_plot)
ggsave(filename = "Final_plots/denovoWES.png",denovoWES_plot, width = 15, height = 7)
denovogenesWES_plot <- ggplot(data = common_results, mapping = aes(x = Family, fill = Gene)) + geom_bar() + theme_minimal() + scale_fill_viridis_d() + labs(title="Figure S31. Genes from De novo insertions detected in WES trios",y = "Number of De novo detected", caption = "Source: Vall d'Hebron WES", fill = "Genes")
print(denovogenesWES_plot)
ggsave(filename = "Final_plots/denovogenesWES.png",denovogenesWES_plot, width = 15, height = 7)
```

## Long reads results

```{r, message=FALSE}
data <- read_csv("rmetl/Longreads_rMETL_final.csv")  %>% distinct()
rMETL_plot <- ggplot(data = data, aes(x = Sample, fill = Type)) + geom_bar() + labs(title = "Figure S31: Number of type of TEs detected by rMETL", y = "Number of TEs", caption = "Source: Long-Read Sample") + theme_minimal() + scale_fill_viridis_d() 
print(rMETL_plot)
ggsave(filename = "Final_plots/numberTEsrMETL.png",rMETL_plot, width = 15, height = 7)
data <- read_csv("rmetl/exp_obs_gene_longread_rMETL.csv")
ggplot(data = data, aes(x = Sample, fill = Distance_Tag)) + geom_bar() + labs(title = "Number of tag distance - rMETL",y = "Number of TEs",caption = "Source:  Long-Read Sample") + theme_minimal() + scale_fill_viridis_d() 
```
```{r, message=FALSE}
data <- read_csv("rmetl/Longreads_scramble_final.csv")  %>% distinct()
rMETL_plot <- ggplot(data = data, aes(x = Sample, fill = Type)) + geom_bar() + labs(title = "Figure S32: Number of type of TEs detected by SCRAMble", y = "Number of TEs", caption = "Source: Long-Read Sample") + theme_minimal() + scale_fill_viridis_d() 
print(rMETL_plot)
ggsave(filename = "Final_plots/numberTEsscramble.png",rMETL_plot, width = 15, height = 7)
data <- read_csv("rmetl/exp_obs_gene_longreads_scramble.csv")
ggplot(data = data, aes(x = Sample, fill = Distance_Tag)) + geom_bar() + labs(title = "Number of tag distance - SCRAMble",y = "Number of TEs",caption = "Source:  Long-Read Sample") + theme_minimal() + scale_fill_viridis_d() 
```

```{r}
data1 <- read_csv("rmetl/exp_obs_gene_longread_rMETL.csv")  %>% distinct()
data2 <- read_csv("rmetl/exp_obs_gene_longreads_scramble.csv")  %>% distinct()

common_rows <- semi_join(data2, data1, by = c("Chromosome","Gene"))
grid.newpage()
  
venn_diagram <-draw.pairwise.venn(
    area1 = nrow(data1),                      
    area2 = nrow(data2),                     
    cross.area = nrow(common_rows),                      
    category = c("rMETL", "SCRAMble"),                 
    fill = c("lightblue", "pink"),       
    lty = "blank",                              
    cat.pos = c(-20, 20),                     
    scaled = TRUE,                          
    cat.fontface = "bold",
    output=TRUE
  )
grid.draw(venn_diagram)
dev.off()
#ggsave(filename = "Final_plots/comparisonrMETL-scramble.png",rMETL_plot, width = 15, height = 7)
```